services/proxy/pkg/middleware/authentication.go

```diff
@@ -5,55 +5,133 @@ import (
 	"net/http"
 	"regexp"
 	"strings"
+
+	"github.com/owncloud/ocis/v2/services/proxy/pkg/webdav"
 )
 
 var (
 	// SupportedAuthStrategies stores configured challenges.
 	SupportedAuthStrategies []string
 
 	// ProxyWwwAuthenticate is a list of endpoints that do not rely on reva underlying authentication, such as ocs.
-	// services that fallback to reva authentication are declared in the "frontend" command on oCIS. It is a list of strings
-	// to be regexp compiled.
-	ProxyWwwAuthenticate = []string{"/ocs/v[12].php/cloud/"}
+	// services that fallback to reva authentication are declared in the "frontend" command on oCIS. It is a list of
+	// regexp.Regexp which are safe to use concurrently.
+	ProxyWwwAuthenticate = []regexp.Regexp{*regexp.MustCompile("/ocs/v[12].php/cloud/")}
+
+	_publicPaths = [...]string{
```

Comment:
Curious: Why are we using arrays instead of slices?

Reply:
It's just my personal style, I have no concrete reason except that you can't accidentally add an item to that array.
See https://go.dev/play/p/n1dVFhXwTWE

Reply:
Fine for me. I thought you might be using the fact that arrays are (in difference to slices) values instead of pointers. So changing a copy won't change the original array. See here for example: https://go.dev/play/p/7Rq_fmJSRVn :smile: